pkg/plugins/golang/v2/api.go

```diff
@@ -176,8 +181,11 @@ func (p *createAPISubcommand) GetScaffolder() (cmdutil.Scaffolder, error) {
 	}
 
 	// Create the actual resource from the resource options
-	res := p.resource.NewResource(p.config, p.doResource)
-	return scaffolds.NewAPIScaffolder(p.config, string(bp), res, p.doResource, p.doController, plugins), nil
+	res := p.options.NewResource(p.config)
+	if !strings.HasPrefix(res.Path, "k8s.io") {
+		res.Domain = p.config.Domain
+	}
```

Comment:
```suggestion
```
This file is the view layer than we ought to not manipulate it here. why the logic was added here and is not done in https://github.com/kubernetes-sigs/kubebuilder/pull/1869/files#diff-4aaa6a94bff36e173cbef610e90660915f5f2dca3fab8860998d370a99cef112R249?

Reply:
There was starting to be too much project v2 specific logic (not accepting empty groups, using only the group for the GVK instead of the qualified group when project version 2, their own set of test cases, ...) in `Options` so I created their own `v2.Options` to keep `golang.Options` clean.